### PR TITLE
(FM-7550) conversion of existing cisco test cases for dual mode

### PR DESCRIPTION
**Pull request by Thomas-Franklin**

### CODE REVIEW

File: tests/beaker_tests/cisco_vtp/test_vtp.rb [hunk 1]
```diff
@@ -0,0 +1,102 @@
+###############################################################################
+# Copyright (c) 2015 Cisco and/or its affiliates.
```
Comment (willmeek) [inline tests/beaker_tests/cisco_vtp/test_vtp.rb]: Change copyright date to 2018

File: tests/beaker_tests/cisco_vxlan_vtep/test_vxlan_vtep.rb [hunk 2]
```diff
@@ -27,6 +27,7 @@
 tests = {
```
Comment (willmeek) [inline tests/beaker_tests/cisco_vxlan_vtep/test_vxlan_vtep.rb:27]: As this is an existing file

` # Copyright (c) 2015 - 2018 Cisco and/or its affiliates.`

File: tests/beaker_tests/cisco_vxlan_vtep_vni/test_vxlan_vtep_vni.rb [hunk 3]
```diff
@@ -59,6 +59,7 @@
 tests = {
   master:        master,
   agent:         agent,
+  sut_platform:  platform,
   platform:      'n(3k-f|5k|6k|7k|9k)',
```
Comment (willmeek) [inline tests/beaker_tests/cisco_vxlan_vtep_vni/test_vxlan_vtep_vni.rb]: If we are setting platform to be

`    platform = tests[:sut_platform]`

Do we need it set here?

File: tests/beaker_tests/lib/utilitylib.rb [hunk 4]
```diff
@@ -1297,7 +1298,13 @@ def platform
 @fretta_slot = nil
 def fretta?(reset_cache=false)
   return @fretta_slot unless @fretta_slot.nil? || reset_cache
-  data = on(agent, facter_cmd('-p cisco.inventory')).output.split("\n")
+  if agent
+    data = on(agent, facter_cmd('-p cisco.inventory')).output.split("\n")
+  else
+    data = `#{AGENTLESS_COMMAND} --facts`
+    inventory_facts = data.match(%r{inventory.*(\n.*)*?(?=virtual_service)})
```
Comment (willmeek) [inline tests/beaker_tests/lib/utilitylib.rb]: Nice inventory parsing (thumbsup)

**Review by willmeek — COMMENTED** [review]
A great PR a couple of minor things

File: tests/beaker_tests/cisco_vxlan_vtep_vni/test_vxlan_vtep_vni.rb (hunk 3 above)
Comment (Thomas-Franklin) [reply tests/beaker_tests/cisco_vxlan_vtep_vni/test_vxlan_vtep_vni.rb]: Yea this was to account for accessing the platform within the test harness classes.

**Comment by Thomas-Franklin** [comment]
Prioritising #51 being merged first. Then will rebase and fix this up for review.

**Comment by willmeek** [comment]
@Thomas-Franklin Are you using this as a branch for all your test conversions, or would you like a merge?

**Comment by Thomas-Franklin** [comment]
> @Thomas-Franklin Are you using this as a branch for all your test conversions, or would you like a merge?

@willmeek happy for this to be merged whenever, was just adding too it really

**Comment by Thomas-Franklin** [comment]
Actually just need to fix up a test case. Will let you know when it is ready @willmeek :-)

**Comment by Thomas-Franklin** [comment]
@willmeek should be ready for review now

**Review by willmeek — APPROVED** [review]
Small Rubocop whitespace fix then good to go

**Pull request merged**